addons/mail/static/src/core/web/chatter.xml

```diff
@@ -135,7 +141,23 @@
                     </FileUploader>
                 </div>
             </div>
-            <Thread t-if="props.hasMessageList" thread="state.thread" t-key="state.thread.localId" order="'desc'" hasScrollAdjust="!props.hasMessageListScrollAdjust" jumpPresent="state.jumpThreadPresent"/>
+            <t t-if="props.hasMessageList">
+                <ChatterSearch
+                    t-if="state.isSearchOpen"
+                    messages="state.searchMessages"
+                    thread="state.thread"
+                    jumpCallback.bind="closeSearch"
+                    searchTerm="state.searchTerm"
+                />
+                <Thread
+                    t-if="!state.isSearchOpen or !state.searchMessages.length"
+                    thread="state.thread"
+                    t-key="state.thread.localId"
+                    order="'desc'"
+                    scrollRef="rootRef"
+                    jumpPresent="state.jumpThreadPresent"
+                />
```

Comment:
Why not reuse `SearchMessagesPanel` in chatter?

At a glance, it's as if there was a desire to keep message in place and give the impression that highlight is added on the messages. However this doesn't really work, because messages are in a card format, and there jump link, etc.

I think it's best to reuse the exact same panel (much easier to reuse working component), and the different design is actually desirable, so that it's very clear visually that one is message listing without search and the other one is on searched messages. (search input is a bit too small).

Besides, if I get the intention was to just highlight the words in messages, this approach requires reusing the exact same components and just pass (modifiers) as props. But I don't think this is a better UI/UX over just reusing the `MessageSearchPanel`.

Let me know if there was another reason to define `ChatterSearch` that I missed!